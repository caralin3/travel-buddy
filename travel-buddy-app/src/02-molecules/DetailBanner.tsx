import React from 'react';
import { formatDate, getDaysUntil } from '../utils';

export interface DetailBannerProps {
  description?: string;
  endDate: string;
  startDate: string;
  title: string;
}

export const DetailBanner: React.FC<DetailBannerProps> = ({ description, endDate, title, startDate }) => (
  <div className="detail-banner bg-secondary text-white px-5 py-4">
    <div className="detail-banner__text d-flex flex-column align-items-start">
      <h1>{title}</h1>
      {!!description && <p dangerouslySetInnerHTML={{ __html: description }} />}
      <p className="h4">
        {formatDate(startDate)} - {formatDate(endDate)}
      </p>
    </div>
    <p className="detail-banner__until circle text-uppercase m-0">
      <strong>{getDaysUntil(startDate)}</strong>
      <strong>days left</strong>
    </p>
  </div>
);
