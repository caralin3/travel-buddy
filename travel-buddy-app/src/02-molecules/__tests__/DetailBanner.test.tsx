import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DetailBanner } from '../DetailBanner';

describe('DetailBanner', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <DetailBanner description="Description" endDate="2022-03-31" startDate="2022-03-20" title="Title" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
