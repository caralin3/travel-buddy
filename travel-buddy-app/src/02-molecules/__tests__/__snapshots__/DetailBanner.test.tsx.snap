// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailBanner should render correctly 1`] = `
<div
  className="detail-banner bg-secondary px-5 py-4"
>
  <div
    className="detail-banner__text d-flex flex-column align-items-start text-white"
  >
    <h1>
      Title
    </h1>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "Description",
        }
      }
    />
    <p
      className="h4"
    >
      March 20, 2022
       - 
      March 31, 2022
    </p>
  </div>
  <p
    className="detail-banner__until circle text-uppercase m-0"
  >
    <strong>
      0
    </strong>
    <strong>
      days left
    </strong>
  </p>
</div>
`;
